docker/local-development.md

```diff
@@ -0,0 +1,279 @@
+      # in addition to the files and directories specified in the root
+      # docker-compose.yml
+      - ...
+    command: pytest -sxv
+```
+
+### 5. `docker-compose.db-ops.yml` (Optional)
+
+Data-rich applications may come with a multi-step process for loading in data.
+In those instances, it can be helpful to define a service that automates each
+of those steps, so initializing your application is a breeze.
+
+Define this, in _addition_ to a `migration` service, because you will always
+want to capture migrations on app start, but it may not always be desirable to
+run the rest of your data loading steps. See [Run custom commands on containers](#run-custom-commands-on-containers) if you want to run arbitrary commands.
+
+```yaml
+version: '2.4'
+
+services:
+  dbload:
+    container_name: <APP_NAME>-dbload
+    image: <APP_NAME>:latest
+    depends_on:
+      - app
+    volumes:
+      - .:/app
+      - ${PWD}/<YOUR_PROJECT>/settings_deployment.py.example:/app/<YOUR_PROJECT>/settings_deployment.py
+    # For some reason, Python logs are buffered if they don't come through
+    # logging. For instant reporting of non-logging output, i.e., print state-
+    # ments or writing to STDOUT directly, run python commands with the -u flag.
+    # https://github.com/moby/moby/issues/12447#issuecomment-263846539
+    command: >
+      bash -c "python manage.py migrate &&
+      python -u manage.py import_data &&
+      ..."
+```
+
+## Using your `docker-compose` setup
+
+### Run the application
+
+If you've defined a `dbload` service, run that first.
+
+```bash
+docker-compose -f docker-compose.yml -f docker-compose.db-ops.yml run --rm dbload
+```
+
+Then, run your application:
+
+```bash
+docker-compose up
+```
+
+Meanwhile, run the tests like:
+
+```bash
+docker-compose -f docker-compose.yml -f tests/docker-compose.yml run --rm app
+```
+
+Stopping your application or test execution is as simple as:
+
+```bash
+docker-compose down
+```
+
+Prefer `docker-compose down` to manually stopping containers one by one, as it
+will stop and clean up all of your containers for you.
+
+### Run custom commands on containers
+
+Sometimes, you'll want to run an individual command on your container, e.g.,
+`update_index` or some data loading operation. Other times, you'll want to run
+your application or tests such that you can drop into a pdb shell.
+
+Luckily, that's easy with `docker-compose`. Simply run a command that looks like
+this:
+
+```bash
+# The --rm flag will remove the temporary container created to run your command after the command exits.
+docker-compose run --rm <CONTAINER_NAME> <COMMAND>
+```
+
+Say you put a pdb breakpoint in your code to debug a problem. `docker-compose up`
+will interpret the pdb breakpoint as a failure and restart your application.
+Instead, run your (Django) application, like so:
+
+```bash
+docker-compose run --rm --service-ports app python manage.py runserver 0.0.0.0:8000
+```
```

Comment:
What do you think about replacing this instruction with the `docker attach` workflow that I linked to above? It requires opening up an extra shell, but it means that you don't need to stop your services and restart them with `docker-compose run` in order to drop into a debugger.